src/pkg/oci/push.go

```diff
@@ -36,154 +34,58 @@ type ConfigPartial struct {
 }
 
 // PushLayer pushes the given layer (bytes) to the remote repository.
-func (o *OrasRemote) PushLayer(b []byte, mediaType string) (ocispec.Descriptor, error) {
+func (o *OrasRemote) PushLayer(ctx context.Context, b []byte, mediaType string) (*ocispec.Descriptor, error) {
 	desc := content.NewDescriptorFromBytes(mediaType, b)
-	return desc, o.repo.Push(o.ctx, desc, bytes.NewReader(b))
+	return &desc, o.repo.Push(ctx, desc, bytes.NewReader(b))
 }
 
-// pushManifestConfigFromMetadata pushes the manifest config with metadata to the remote repository.
-func (o *OrasRemote) pushManifestConfigFromMetadata(metadata *types.ZarfMetadata, build *types.ZarfBuildData) (ocispec.Descriptor, error) {
-	annotations := map[string]string{
-		ocispec.AnnotationTitle:       metadata.Name,
-		ocispec.AnnotationDescription: metadata.Description,
+// PushManifestConfigFromMetadata pushes the manifest config with metadata to the remote repository.
+func (o *OrasRemote) PushManifestConfigFromMetadata(ctx context.Context, annotations map[string]string, configMediaType string) (*ocispec.Descriptor, error) {
+	if annotations[ocispec.AnnotationTitle] == "" {
+		return nil, fmt.Errorf("invalid annotations: please include value for %q", ocispec.AnnotationTitle)
```

Comment:
note: will need to communicate to UDS CLI that they will need to populate this annotation, and show how to make a similar function to the one in Zarf.